<a href="https://colab.research.google.com/github/alif23-blip/pandas-dan-numpy/blob/main/pandas_%26_numpy.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np

harga = np.array([5000, 7000, 3000, 12000, 4500])
print('rata-rata harga:', harga.mean())
print('harga tertinggi:', harga.max())
print('harga setelah diskon 10%', harga * 0.9)

rata-rata harga: 6300.0
harga tertinggi: 12000
harga setelah diskon 10% [ 4500.  6300.  2700. 10800.  4050.]


analisis 1

Jika harga berupa **list Python biasa** seperti [5000, 7000, 3000, 12000, 4500] lalu dikalikan 0.9,
maka akan muncul **TypeError**, karena operator `*` pada list Python hanya bisa digunakan dengan
bilangan bulat (int) untuk mengulang/menduplikasi isi list, bukan untuk mengalikan tiap elemennya.
Karena 0.9 adalah bilangan desimal (float), Python tidak tahu cara "mengulang" list dengan angka
pecahan

In [ ]:
import pandas as pd

data_kantin = {
    'menu': ['nasi goreng', 'es teh', 'mie ayam', 'es teh', None],
    'harga': [12000, 4000, 10000, 4000, 8000],
    'terjual': [23, 40, None, 35, 18]
}
df = pd.DataFrame(data_kantin)
print(df)

          menu  harga  terjual
0  nasi goreng  12000     23.0
1       es teh   4000     40.0
2     mie ayam  10000      NaN
3       es teh   4000     35.0
4         None   8000     18.0


analisis 2

kolom yang memiliki data kosong adalah menu dan terjual, jika data langsung dianalisis dan tidak dibersihkan, maka hasil analisis akan tidak akurat

In [ ]:
df = pd.read_csv('data_kantin - data_kantin.csv')

print(df.head())
print(df.info())
print(df.describe())
print(df.shape)

      tanggal         menu kategori  harga  terjual
0  2026-08-20   Teh Hangat  Minuman   3000     11.0
1  2026-08-31  Nasi Goreng  Makanan  12000     29.0
2  2026-08-21  Nasi Goreng  Makanan  12000     28.0
3  2026-08-28   Teh Hangat  Minuman   3000     12.0
4  2026-08-21     Es Jeruk  Minuman   5000     19.0
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    object 
 1   menu      107 non-null    object 
 2   kategori  110 non-null    object 
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 4.4+ KB
None
              harga     terjual
count    110.000000  104.000000
mean    6336.363636   22.923077
std     3520.001232   10.831178
min     2000.000000    8.000000
25%     3000.000000   13.000000
50%     5000.000000   20.000000
75%    10000.000000 

analisis 3

kolom yang jumlah non-null yang lebih sedikit adalah menu dan terjual, berarti terdapat data yang kosong pada kolom menu dan terjual




In [ ]:
print(df.isnull().sum()) # jumlah data kosong tiap kolom

df['terjual'] = df['terjual'].fillna(0) # isi kekosongan dengan 0
df = df.dropna(subset=['menu']) # hapus baris jika kolom menu kosong

tanggal     0
menu        3
kategori    0
harga       0
terjual     6
dtype: int64


analisis 4

kolom terjual diisi (fillna) karena isi kolom terjual berupa *integer* yang dimana data yang kosong dapat dicari dengan nilai yang ditentukan, lalu kolom menu diisi (dropna) karena isi kolom menu berupa *string* yang membuat nilai yang kosong tidak dapat dicari dengan nilai yang ditentukan, oleh karena itu lebih aman dengan menghapus data dari kolom menu yang kosong


In [ ]:
print(df.duplicated().sum())
df = df.drop_duplicates()

df['harga'] = df['harga'].astype(int)
print (df.dtypes)

5
tanggal      object
menu         object
kategori     object
harga         int64
terjual     float64
dtype: object


/tmp/ipykernel_1328/1134032328.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['harga'] = df['harga'].astype(int)


Dataset berkurang dari 107 baris → 102 baris, memastikan dtypes benar sebelum analisis itu penting supaya operasi berjalan sesuai yang diharapkan dan hasil analisis tidak salah atau error di tengah jalan

In [ ]:
laris = df[df['terjual'] > 20]
urut = df.sort_values(by='terjual', ascending=False)

df['total_pendapatan'] = df['harga'] * df['terjual']

ringkasan = df.groupby('menu')['total_pendapatan'].sum()
print(ringkasan)

menu
Es Jeruk       1460000.0
Es Teh         2144000.0
Kerupuk         972000.0
Mie Ayam       2410000.0
Nasi Goreng    3924000.0
Roti Bakar     1288000.0
Teh Hangat      459000.0
Name: total_pendapatan, dtype: float64


/tmp/ipykernel_1328/2047603725.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['total_pendapatan'] = df['harga'] * df['terjual']


In [ ]:
menu dengan total pendapatan tertinggi adalah nasi goreng, yaitu Rp4.188.000

membantu penjual untuk melihat menu yang paling laris yang membuat penjula dapat menambah stok ke menu yang laris tersebut